R/missings.R

```diff
@@ -12,16 +12,40 @@
 #' @param case_weights A vector of the same length as `truth` and `estimate`, or
 #'   `NULL` if case weights are not being used.
 #'
+#' @param censoring_weights A vector of the same length as `truth` and
+#'   `estimate`, or `NULL` if censoring weights are not being used. Defaults to
+#'   `NULL`.
+#'
+#' @param eval_times A vector of the same length as `truth` and
+#'   `estimate`, or `NULL` if time points are not being used. Defaults to
+#'   `NULL`.
+#'
 #' @seealso [metric-summarizers]
 #'
 #' @name yardstick_remove_missing
 NULL
 
 #' @rdname yardstick_remove_missing
 #' @export
-yardstick_remove_missing <- function(truth, estimate, case_weights) {
-  complete_cases <- stats::complete.cases(truth, estimate, case_weights)
-  truth <- truth[complete_cases]
+yardstick_remove_missing <- function(truth,
+                                     estimate,
+                                     case_weights,
+                                     censoring_weights = NULL,
+                                     eval_times = NULL) {
+  complete_cases <- stats::complete.cases(
+    truth, estimate, case_weights, censoring_weights, eval_times
+  )
+
+  if (inherits(truth, "Surv")) {
+    Surv_type <- attr(truth, "type")
+
+    truth <- truth[complete_cases, ]
+
+    attr(truth, "type") <- Surv_type
```

Comment:
They make it so that subsetting still works even if survival isn't loaded

``` r
library(yardstick)

data("lung_surv")

lung_surv$surv_obj[1:10, ]
#>       time status
#>  [1,]  455      1
#>  [2,]  455      1
#>  [3,]  455      1
#>  [4,]  210      1
#>  [5,]  210      1
#>  [6,]  210      1
#>  [7,] 1022      0
#>  [8,] 1022      0
#>  [9,] 1022      0
#> [10,]  310      1

library(survival)

lung_surv$surv_obj[1:10, ]
#>  [1]  455   455   455   210   210   210  1022+ 1022+ 1022+  310
```

<sup>Created on 2023-02-24 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>